### ISSUE TITLE
`format_quantity` function missing to convert Decimal values in to K8s quantity string

### ISSUE THREAD

**Issue by rkschamer** [kubernetes-client/python#2205]
**What is the feature and why do you need it**:

There is [`parse_quantity`](https://github.com/kubernetes-client/python/blob/master/kubernetes/utils/quantity.py) to convert a canonical K8s quantity into a `Decimal` value. However, an opposite function, like `format_quantity` is missing, making it hard to transform a Decimal into a K8s quantity. 

My use case was that I needed to load the resource requirements of serval pods, finding the maximum value and setting it for all pods. Due to potentially different SI suffixes, I've used `parse_quantity`, however setting the resource requirements for all other Pods, requires me to convert back from a `Decimal` into a canonical K8s quantity. 

**Describe the solution you'd like to see**:

So far, I've implemented the function in my project, but I think that might be something also others could benefit. For this reason, I'd like to open a pull request to contribute this function.
```[tasklist]
### Tasks
```

**Comment by rkschamer** [comment]
Hi @roycaihw,

thanks for looking into my request. I'm sorry that it took me a while to come back to you. 

I'm not entirely sure, on which aspect my help is needed. Maybe it my description for the desired function was not quite clear. So let me try again: 

> There is already utils.parse_quantity which I used in my project, but I also needed a way to convert a Decimal number back into a Kubernetes quantity. I propose to add this functionality to the Python K8s client.

I opened the issue to ask if such a function is desired and since, I've already implemented it for my own project, I want to contribute. To illustrate my request a bit more, I've no created a draft PR for the function, so that you can have a look (there is also an example how to use the function): https://github.com/kubernetes-client/python/pull/2216

Please let me know if you're in favor of the change. Then I'll convert into an actual PR and sign the CLA.

**Comment by roycaihw** [comment]
Hi @rkschamer, the feature sounds good! Please feel free to convert the draft to an actual PR and we can review and merge it

**Comment by rkschamer** [comment]
> Hi @rkschamer, the feature sounds good! Please feel free to convert the draft to an actual PR and we can review and merge it

Thanks, @roycaihw! I've now added tests and converted to an actual PR. Please feel free to review the PR.

**Comment by superlazyname** [comment]
Could this feature be merged in? This seems really helpful.

People on Stack Overflow are suggesting to roll your own converter from '100m' to decimal and back and that's obviously less than ideal from a maintainability standpoint, it seems like only Golang has code in it to convert both ways.

**Comment by rkschamer** [comment]
@superlazyname, I'd be very happy if my PR #2216 could be merged. I'm just waiting for a review from @roycaihw, who is assigned to it.

**Comment by superlazyname** [comment]
Why can't somebody review the PR?

Is this project still accepting community contributions?

**Comment by rkschamer** [comment]
@kubernetes/sig-pr-reviews can you help to review this PR?

/remove-lifecycle rotten

**Comment by rkschamer** [comment]
> Why can't somebody review the PR?
> 
> Is this project still accepting community contributions?
> […](#)

Hi @superlazyname,

let me try to reach out to the community via Slack in the hope to find a reviewer. Message is here: https://kubernetes.slack.com/archives/C09R23FHP/p1738161320260709

**Comment by psnilesh** [comment]
When will this release ?
